fakegreet/src/main.rs

```diff
@@ -151,12 +151,21 @@ async fn client_handler(ctx: &Context, mut s: UnixStream) -> Result<(), Error> {
 }
 
 pub async fn server() -> Result<(), Error> {
-    let path = "./greetd.sock";
-    std::env::set_var("GREETD_SOCK", path);
-
-    let _ = std::fs::remove_file(path);
+    let mut path = env::current_dir()?;
+    let current_dir = format!(
+        "{}/greetd.sock",
+        std::fs::canonicalize(&path)
+            .unwrap()
+            .to_string_lossy()
+            .to_string(),
```

Comment:
```suggestion
            .to_str()
            .unwrap()
```

We don't need an owned string, and `to_string_lossy` may return a string that differs from the input.

Reply:
Ok, I have made the change